### PR TITLE
[secret-copier] Add secret creation validation

### PR DESCRIPTION
**Pull request by meatuses**
## Description
Add validation through `ValidatingAdmissionPolicy` to prohibit manual creation of secrets with label `secret-copier.deckhouse.io/enabled` outside of `default` namespace. 

## Why do we need it, and what problem does it solve?
This would help with accidental creation of secrets, including automatic creation with wrong `secretTemplates`, because such secrets would be deleted by `secret-copier` anyway.

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: secret-copier
type: chore
summary: Add validation of secrets with secret-copier labels
impact_level: low
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @meatuses. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @meatuses. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @meatuses. I've started the workflow for label 'e2e/run/aws'...

File: modules/600-secret-copier/templates/validation.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+spec:
+  auditAnnotations:
+    - key: source-user
+      valueExpression: '''User: '' + string(request.userInfo.username) + '' tries to
+      create secret with `secret-copier.deckhouse.io/enabled` label outside of `default` namespace'''
+  failurePolicy: Fail
+  matchConstraints:
+    matchPolicy: Equivalent
+    namespaceSelector:
+      matchExpressions:
```
Comment (yalosev) [inline modules/600-secret-copier/templates/validation.yml:17]: I guess it's better to place this selector in the ValidatingAdmissionPolicyBinding

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @meatuses. I've started the workflow for label 'e2e/run/yandex-cloud'...

File: modules/600-secret-copier/templates/validation.yml (hunk 1 above)
Comment (meatuses) [reply modules/600-secret-copier/templates/validation.yml]: moved namespaceselector to the binding resource

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @meatuses. I've started the workflow for label 'e2e/run/aws'...

**Comment by alexey-gavrilov-flant** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.27 11346274868 11346274868-1-con-1-27 /sys/deckhouse-oss/install:pr10066 redos@84.252.130.229

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @alexey-gavrilov-flant. I've recognized your '/e2e/abort' command and started the workflow...

**Review by yalosev — APPROVED** [review]

**Pull request merged**